src/System.IO.Pipelines/src/System/IO/Pipelines/PipeWriter.cs

```diff
@@ -51,5 +51,44 @@ public virtual ValueTask<FlushResult> WriteAsync(ReadOnlyMemory<byte> source, Ca
             this.Write(source.Span);
             return FlushAsync(cancellationToken);
         }
+
+        /// <summary>
+        /// Asynchronously reads the bytes from the specified stream and writes them to the <see cref="PipeWriter"/>.
+        /// </summary>
+        /// <param name="source">The stream from which the contents will be copied.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A task that represents the asynchronous copy operation.</returns>
+        protected internal virtual async Task CopyFromAsync(Stream source, CancellationToken cancellationToken = default)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                Memory<byte> buffer = GetMemory();
```

Comment:
Do we want to pass in a sizeHint here/make it configurable?

Reply:
See https://github.com/dotnet/corefx/issues/27246#issuecomment-462074106